src/reports/reportData.ts

```diff
@@ -0,0 +1,77 @@
+import * as systeminformation from "systeminformation";
+
+export type OsInfo = {
+  platform: string;
+  distro: string;
+  arch: string;
+  kernel: string;
+};
+
+export type CpuInfo = {
+  manufacturer: string;
+  brand: string;
+  speedGHz: number;
+  cores: number;
+};
+
+export type Specs = {
+  os: OsInfo;
+  cpu: CpuInfo;
+  memoryBytes: number;
+};
+
+export type ReportData = {
+  timestamp: string;
+  os: string;
+  kernel: string;
+  cpu: string;
+  cores: string;
+  speedGHz: string;
+  memoryBytes: string;
+};
+
+let specs: Promise<Specs> | undefined;
+
+async function getSpecs(): Promise<Specs> {
+  if (!specs) {
+    specs = new Promise<Specs>((resolve) => {
+      void systeminformation.cpu().then((cpuData) => {
+        void systeminformation.osInfo().then((osData) => {
+          void systeminformation.mem().then((memoryData) => {
+            resolve({
```

Comment:
You can write this as:
```typescript
    specs = systeminformation.cpu().then((cpuData) => 
         systeminformation.osInfo().then((osData) => 
          systeminformation.mem().then((memoryData) => 
            ({
                os: {
                ...
            })
            )
            );
```

because now you didn't use `awaits` and promise that returns a promise is chained. See: https://javascript.info/promise-chaining